frame/composable-support/src/math/safe.rs

```diff
@@ -1,18 +1,27 @@
 use sp_runtime::{
 	helpers_128bit::multiply_by_rational,
-	traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, One, Saturating, Zero},
+	traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, Zero},
 	ArithmeticError,
 };
 
-/// Thin convenient wrapper to lift the Result in a Dispatch context.
+/// Thin wrapper around [`multiply_by_rational`], transforming any errors into [`ArithmeticError`]
+/// for easier use in pallets and Dispatch-related contexts. See [`multiply_by_rational`] for more
+/// information.
+///
+/// Note: [`multiply_by_rational`] clamps `c` at a minimum of `1`, which can be confusing.
+/// [`safe_multiply_by_rational`] instead returns a divide by zero error if `c == 0`.
 pub fn safe_multiply_by_rational(a: u128, b: u128, c: u128) -> Result<u128, ArithmeticError> {
-	multiply_by_rational(a, b, c).map_err(|_| ArithmeticError::Overflow)
+	if c == 0 {
+		Err(ArithmeticError::DivisionByZero)
+	} else {
+		multiply_by_rational(a, b, c).map_err(|_| ArithmeticError::Overflow)
+	}
 }
 
-/// little bit slower than maximizing performance by knowing constraints.
-/// Example, you sum to negative numbers, can get underflow, so need to check on each add; but if
-/// you have positive number only, you cannot have underflow. Same for other constrains, like non
-/// zero divisor.
+// little bit slower than maximizing performance by knowing constraints.
```

Comment:
can mark TODO to move it into module doc. the idea is that in half of cases can use constrains to avoid safe math, which will be faster in runtime, but riskier to miss issue and get bug